platform/mbed_alloc_wrappers.cpp

```diff
@@ -329,10 +329,125 @@ extern "C" void $Sub$$free(void *ptr) {
 #endif // #if defined(MBED_MEM_TRACING_ENABLED) || defined(MBED_HEAP_STATS_ENABLED)
 
 /******************************************************************************/
-/* Allocation wrappers for other toolchains are not supported yet             */
+/* IAR memory allocation wrappers                                           */
 /******************************************************************************/
+#elif defined(__ICCARM__)
+
```

Comment:
You might be able to combine the ARM and IAR code and just define the function names. Something like this:

```
#elif defined(TOOLCHAIN_ARM)
#define SUPER_MALLOC    $Super$$malloc
#define SUB_MALLOC      $Sub$$malloc
...

#elif defined(__ICCARM__)
#define SUPER_MALLOC    $Super$$__iar_dlmalloc
#define SUB_MALLOC      $Sub$$__iar_dlmalloc
...

#endif

extern "C" {
    void *SUPER_MALLOC(size_t);
    void *SUPER_REALLOC(void *ptr, size_t size);
    void *SUPER_CALLOC(size_t nmemb, size_t size);
    void SUPER_FREE(void *ptr);
}

extern "C" void *SUB_MALLOC(size_t size) {
    void *ptr = NULL;
...
}
...
```

Reply:
I intentionally separated IAR as was not sure if something additional would be required or not. Since its now working can merge ARM and IAR implementation